lib/kuniri/parser/xml_output_format.rb

```diff
@@ -23,26 +23,39 @@ def class_generate(pClass)
       pClass.each do |singleClass|
         @outputEngine.classData name: singleClass.name,
                                 visibility: singleClass.visibility do
+          methods_counter = 0
           wrapper.comment_generate(singleClass)
 
-          unless singleClass.aggregations.empty?
-            wrapper.aggregation_generate(singleClass.aggregations)
-          end
+          wrapper.handle_aggregations(singleClass.aggregations)
           wrapper.inheritance_generate(singleClass.inheritances)
-          unless singleClass.attributes.empty?
-            wrapper.attribute_generate(singleClass.attributes)
-          end
+          wrapper.handle_attributes(singleClass.attributes)
+
           singleClass.constructors.each do |singleConstructor|
             wrapper.function_behaviour_generate('constructorData',
                                                 singleConstructor)
+            methods_counter += 1
```

Comment:
The constructor and methods are two different things. It does not make sense to count a constructor like a method. E.g.: One class can have multiple constructors and multiple methods. Remove the counter from this part, maybe you can send another PR with the constructor part.